src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -313,10 +330,20 @@ public void ShowHint()
         // Show full help
         public void ShowHelp(string commandName = null)
         {
-            var headerBuilder = new StringBuilder("Usage:");
             for (var cmd = this; cmd != null; cmd = cmd.Parent)
             {
                 cmd.IsShowingInformation = true;
+            }
+
+            Console.WriteLine(GetHelpText(commandName));
+        }
+
+        public string GetHelpText(string commandName = null)
+        {
+            var sb = new StringBuilder();
```

Comment:
why do you create it here when you use it only at the very bottom. I am also not sure whether using a StringBuilder is needed here at all - you concatenate just 5 strings and you don't do this repeatedly. Given that this is a command line app I don't think string vs. sb will be even noticeable to the user...